src/main/java/com/amazonaws/encryptionsdk/kms/KmsMasterKeyProvider.java

```diff
@@ -125,7 +125,12 @@ protected KmsMasterKeyProvider(final AWSKMSClient kms, final Region region, fina
         kms_ = kms;
```

Comment:
We should `@Deprecated` the old constructors and encourage folks to move to the builder API.

I'd also suggest a `builder()` static method on `KmsMasterKeyProvider` for consistency with other builders in the SDK.